.github/workflows/test-basic-pwc-failed-only.yml

```diff
@@ -0,0 +1,157 @@
+# Demonstrates a workflow that initially runs the previously failed tests
+## Consists of 3 jobs: previousFailedFirst, fullRun, recordFailedTests
+##
+## previousFailedFirst - downloads previous failures for the branch, if there are any, it runs those failures using a matrix
```

Comment:
How does it know what is "previous"?

Reply:
It uses the github action `dawidd6/action-download-artifact@v6`  which queries the github workflows api. It has a bunch of settings, I do see that I need to update `workflow_conclusion` to `completed`. Then what it does is look at the last completed workflow for the source branch (can be set to a commit sha, or a PR number instead, or a run_id if someone happens to want to reference a specific run ) and downloads the artifacts from that.

Reply:
gotcha, i think it'd be helpful to elaborate on this point.

Reply:
Update the comment with more details